src/run_throttlebot.py

```diff
@@ -1325,6 +1326,51 @@ def filter_mr(mr_allocation, acceptable_resources, acceptable_services, acceptab
 
     return mr_allocation
 
+def set_mr_provision_detect_id_change(redis_db, mr, new_mr_allocation, workload_config):
+    while True:
+        try:
+            resource_modifier.set_mr_provision(mr, new_mr_allocation, workload_config)
+            # print "No error in Container ID. Continuing..."
+            break
+        except SystemError as e:
+            print "SystemError caught. Container ID changed."
+            print "Updating the container_id of mr: " + mr.to_string()
+            update_mr_id(redis_db, mr)
+            pass
+
+def update_mr_id(redis_db, mr_to_change):
```

Comment:
Don't write comments embedded in the code. Please write the logic in a block comment above the function definition.